### ISSUE TITLE
ADLX isn't detecting AMD 7800XT

### ISSUE THREAD

**Issue by motosxorpio** [Rem0o/FanControl.Releases#2604]
New release states built in ADLX AMD support. However, after updating, FanControl no longer can adjust speeds on the AMD gpu. Prior to this update (195) GPU sensor and control was established without issue.

**Comment by Phoenix831** [comment]
Having a similar issue here, with V195 my 6800XT's fans are no longer controllable with this software. Sensors show up as normal but I get a ? when trying to identify and manually control the GPU's fans. There is no other software running on the PC that could take control of the GPU fans. Trying to load the plugin that allowed it to work also produces an error loop, had to revert to v194 to  be able to control my GPU fans via this.

**Comment by CartierCodes** [comment]
When I update the app to V195 with the previous ADLX plugin still installed, it says "ADLX could not initialize or has no sensors". When I remove the plugin it simply doesn't recognize anything related to my GPU.

I have a 7900XTX

**Comment by Phoenix831** [comment]
> When I update the app to V195 with the previous ADLX plugin still installed, it says "ADLX could not initialize or has no sensors". When I remove the plugin it simply doesn't recognize anything related to my GPU.
> 
> I have a 7900XTX

I had to go into the settings. Under Sensors > Sources, there should be a checkbox for ALDX.

**Comment by Rem0o** [comment]
If you want to use the built-in ADLX support, you need to disable/remove the ADLX plugin.

**Comment by CartierCodes** [comment]
> > When I update the app to V195 with the previous ADLX plugin still installed, it says "ADLX could not initialize or has no sensors". When I remove the plugin it simply doesn't recognize anything related to my GPU.
> > I have a 7900XTX
> 
> I had to go into the settings. Under Sensors > Sources, there should be a checkbox for ALDX.

Ok this worked, but the sensors are misnamed. My GPU sensors and integrated graphics sensors names are flipped

**Comment by CartierCodes** [comment]
> If you want to use the built-in ADLX support, you need to disable/remove the ADLX plugin, as the two will conflict.

Ok this worked to get it somewhat usable

**Comment by motosxorpio** [comment]
> If you want to use the built-in ADLX support, you need to disable/remove the ADLX plugin, as the two will conflict.

I have no plugin installed. When I edit sources to add a check in the box for ADLXWrapper, it doesn't "stick", box remains unchecked after FanControl tells me to save configuration and refresh.

**Comment by ben2224** [comment]
Hi, having similar issues where the plugin worked before v195 but isn't working when built in.

**Comment by Rem0o** [comment]
@ben2224  do you have both enabled? You need to pick either the plugin or the built-in.

**Comment by ben2224** [comment]
Only the one showed in the app, but I'm going to do a fresh installation.

**Comment by ben2224** [comment]
With a fresh install none of the amd gpu sensors are found

**Comment by Rem0o** [comment]
You got no control card for your GPU?  If you are talking about the dialog popup when you load your config, if you come from the plugin, it's normal it doesn't find the plugin's sensors.  The new one will be seen as different.

**Comment by ben2224** [comment]
> With a fresh install none of the amd gpu sensors are found

Oddly, installing v194 and the plugin is now not detecting sensors either, so I might have another issue in itself.

**Comment by ben2224** [comment]
> You got no control card for your GPU? If you are talking about the dialog popup when you load your config, if you come from the plugin, it's normal it doesn't find the plugin's sensors. The new one will be seen as different.

Sorry but I'm not sure what you mean. On the settings page it says ADLX - No sensors from plugin detected.

**Comment by Rem0o** [comment]
> > With a fresh install none of the amd gpu sensors are found
> 
> Oddly, installing v194 and the plugin is now not detecting sensors either, so I might have another issue in itself.

Indeed. Can you do fan stuff in Adrenaline? Try a reboot if you haven't already.

**Comment by dimitrovmaksim** [comment]
> My GPU sensors and integrated graphics sensors names are flipped

Same here for my RX7800XT
![image](https://github.com/Rem0o/FanControl.Releases/assets/20456492/992c63dd-d6ad-42c9-8aba-ce76ca469599)

**Comment by ben2224** [comment]
> > > With a fresh install none of the amd gpu sensors are found
> > 
> > 
> > Oddly, installing v194 and the plugin is now not detecting sensors either, so I might have another issue in itself.
> 
> Indeed. Can you do fan stuff in Adrenaline? Try a reboot if you haven't already.

Yes I can control them in Adrenaline, I've done updates and rebooted. Only thing I can thing of is I rolled backed my graphics/adrenaline driver, but the software worked fine before with this same driver.

**Comment by benshalev849** [comment]
Same situation as @dimitrovmaksim  mentioned
The 7900XTX is displayed as "AMD Radeon(TM) Graphics (1)"
![image](https://github.com/Rem0o/FanControl.Releases/assets/87974262/4de16316-1647-46ee-9639-2e1dda3384d6)

I can confirm it's indeed the XTX and not the CPU IGPU by seeing the hotspot on adrenalin (for the 7900xtx) match the hotspot displayed by fancontrol.
A seen here:
![image](https://github.com/Rem0o/FanControl.Releases/assets/87974262/68129e35-7b7a-4c6b-868d-c7409d6b4390)

@Rem0o Do you want us to open another issue?

NOTE: The fans ARE being properly controlled by fancontrol (tested with 100% and 0% to see it works) but just not displayed.

I have removed the ADLX plugin, enabled it in settings, and restarted program.

**Comment by Rem0o** [comment]
@dimitrovmaksim @benshalev849  yes that's a seperate issue.

**Comment by ben2224** [comment]
@Rem0o do you have any idea why ADLX is no longer working for me, I've done a fresh AMD driver install with the factory reset option to clear anything that might be confusing things. 

The log file says "ADLX could not initialize or has no sensors." on both v194 and v195

**Comment by benshalev849** [comment]
> @Rem0o do you have any idea why ADLX is no longer working for me, I've done a fresh AMD driver install with the factory reset option to clear anything that might be confusing things.
> 
> The log file says "ADLX could not initialize or has no sensors." on both v194 and v195

1) remove ADLX plugin
2) upgrade to V195
3) press “continue” when it displays the error
4) go to settings and allow “ADLXWrapper”
5) scan fans (press on toolbar to the left, press “scan fans”
6) make your config and save it
7) restart to make sure it works (or increase gpu fan to 0/100 to see)
8) enjot

**Comment by ben2224** [comment]
> > @Rem0o do you have any idea why ADLX is no longer working for me, I've done a fresh AMD driver install with the factory reset option to clear anything that might be confusing things.
> > The log file says "ADLX could not initialize or has no sensors." on both v194 and v195
> 
> 1. remove ADLX plugin
> 2. upgrade to V195
> 3. press “continue” when it displays the error
> 4. go to settings and allow “ADLXWrapper”
> 5. scan fans (press on toolbar to the left, press “scan fans”
> 6. make your config and save it
> 7. restart to make sure it works (or increase gpu fan to 0/100 to see)
> 8. enjot

I'm not sure what you mean by "scan fans" as I don't have this option, refreshing sensors does not fix anything.

**Comment by motosxorpio** [comment]
> New release states built in ADLX AMD support. However, after updating, FanControl no longer can adjust speeds on the AMD gpu. Prior to this update (195) GPU sensor and control was established without issue.

Update to the above issue. AFTER updating to 196, AND REMOVING any custom fan curves/overclock profiles from Adrenaline Software, AND rebooting twice: New sensor detected for AMD 7800XT with actual control of the fan(s) speeds. YAY! Chipset can be cooled too! Thanks for the update @Rem0o

**Comment by lunaroc** [comment]
For several versions it has been impossible for me to see the rpm of the fans of my 7800 XT, which I could see in previous versions.

I don't need to have control of the GPU fans. It works for me as it did in previous versions where I could only see them.

Please, is there any way to get versions prior to the ADLX placement?
I understand that v193 or earlier. But I can't find where I can download them.

Thanks for the work.

**Comment by Rem0o** [comment]
@lunaroc  
![image](https://github.com/user-attachments/assets/0aaf1872-fc81-40f9-b9c9-58c630d6b355)

**Comment by lunaroc** [comment]
@Rem0o 
![Captura de pantalla 2024-07-13 024954](https://github.com/user-attachments/assets/91aac881-05af-4136-a2aa-980dff041266)
![Captura de pantalla 2024-07-13 024729](https://github.com/user-attachments/assets/77684ee7-d477-43bf-90d7-9d763528c565)
![Captura de pantalla 2024-07-13 024711](https://github.com/user-attachments/assets/f2cf0986-e793-48a2-8678-bb15fdc400bf)
![Captura de pantalla 2024-07-13 024921](https://github.com/user-attachments/assets/65ddc94f-cc38-47af-bcca-292562a28495)

**Comment by Rem0o** [comment]
@lunaroc  any logs.txt?

**Comment by Rem0o** [comment]
Also, access to this? 
![image](https://github.com/user-attachments/assets/9cfaa925-e461-48b7-862f-659718f6bfaf)

**Comment by lunaroc** [comment]
Hi @Rem0o, 
First of all, thank you for your help. I attach a screenshot where the Log.txt appears.
The same error message always keeps appearing.

 And another worrying error is that if this message appears, the fancontrol program does not start automatically and I have to initialize it manually. And I understand that this is very dangerous if you are going to give full control of the GPU fans to this program.

As I told you, I would really appreciate it if you could send me version 193 or earlier where the FANCONTROL program could easily see the RPM of the GPU fans without having to act on its operation or configuration of its adrenalin software.
thank you so much.
![Captura de pantalla 2024-07-13 191123](https://github.com/user-attachments/assets/940f9811-21a5-4eba-9187-a91dd7dee0d9)

**Comment by Rem0o** [comment]
FYI all latest testing I've done was on 
![image](https://github.com/user-attachments/assets/684acd11-3e87-4af1-ba4f-d84ad0c10842)

**Comment by lunaroc** [comment]
Sorr @Rem0o, but I had not read anywhere about the requirement to have version 24.6.1 of the AMD driver, especially since I have been using Fancontrol for many months with much older versions.

If it is impossible to use fancontrol v196  with the 7800xt with the driver version prior to 24.6.1, there is an unsolvable problem.

the  version 24.6.1 is very very broken. It freezes and causes FPS losses in multiple games, and the GPU consumption goes from 60W with previous versions to being above 70W all the time.

This version of AMD driver 24.6.1 is not going well in many aspects.

#https://community.amd.com/t5/drivers-software/stuttering-issues-with-driver-24-6-1-killing-fps/m-p/693185
#https://www.reddit.com/r/AMDHelp/comments/1dxh332/2461_drivers/
#https://www.reddit.com/r/radeon/comments/1dptumf/amd_software_adrenalin_edition_2461_release_notes/
...

**Comment by Rem0o** [comment]
Did a driver rollback and I can confirm on 24.5.1, ADLX doesn't return any available GPU to control. Seems like a driver bug, since once I reinstalled 24.6.1 everything works as expected.

**Comment by lunaroc** [comment]
> Did a driver rollback and I can confirm on 24.5.1, ADLX doesn't return any available GPU to control. Seems like a driver bug, since once I reinstalled 24.6.1 everything works as expected, so I'm sorry to say we'll have to wait for a driver update if 24.6.1 is broken for games.

Well, we will have to wait for them to update the AMD driver

@Rem0o , one question: before using ADLX control, in previous versions, from FANCONTROL, it was possible to see the GPU fan RPM, and I think the GPU temperature was also displayed. Is it very difficult to implement this option so we don't have to rely on external drivers from a specific company?

But as I said before, Fancontrol is the best for controlling fans, but as a programmer, you know that the likelihood of having problems is proportional to the complexity of the program.

**Comment by Rem0o** [comment]
@lunaroc  it ain't external drivers, it's the official one provided by AMD. The other provider is the "GPU" checkbox of LibrehardwareMonitor.  It uses old ADL calls, it can see a few stuff but can't control the fan.

**Comment by lunaroc** [comment]
Sorry for my English, that's what I've been trying to explain from the beginning.In previous versions of FANCONTROL, before you implemented the ADLX API, the LHM option (GPU) did provide information about GPU fan speed and temperature. Unfortunately, even with the GPU box in LHM checked, no GPU sensor appears.

![imagen](https://github.com/user-attachments/assets/0cb966c9-6b7f-4b9b-aad8-30a27d8bb742)
![imagen](https://github.com/user-attachments/assets/4988ce81-902b-4c82-b057-9051ce2a02b1)
![imagen](https://github.com/user-attachments/assets/55a5cb77-49a8-444c-bc12-e06a3aea494d)

Currently, in V196, no API (ADLX or LHM) provides information about the GPU. That's what kills me.
I don't need to control the GPU fans, just to have some data from the GPU (temperature or fan RPM). And no API (ADLX or LHM) provides any information about the GPU.
I don't need to control the GPU fans, just to have some data from the GPU (temperature or fan RPM).

Sorry again for my English.

**Comment by Rem0o** [comment]
Weird as nothing changed on the LHM side.  If you fire up the LHM program separately, do you get GPU sensors?
https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/releases/tag/v0.9.3

Otherwise all previously available versions are available here: https://github.com/Rem0o/FanControl.Releases/releases

**Comment by lunaroc** [comment]
Hello @Rem0o ,
After conducting multiple tests, thanks to your previous post about the versions of LHM and Fancontrol, I was able to identify the problem.
The problem does not come from LHM or Fancontrol.
The issue with the disappearing sensors arises when you try to downgrade AMD Software: Adrenalin Edition. (It happens both from 24.6.1 to 24.5.1, and from 25.5.1 to 24.4.1).

The following happened to me: After downgrading from 24.6.1 to version 24.5.1, as I told you, neither LHM provided any information about the 7800 XT GPU.
![Captura de pantalla 2024-07-17 220850](https://github.com/user-attachments/assets/6e0400db-85c4-4a2f-9101-a8f14b70b770)
After performing a clean installation of 24.5.1 (all previous drivers deleted) and even downgrading to 24.4.1, the problem persisted. LHM did not provide any information about the 7800 XT GPU.

In looking for ways to remove all traces of the driver, the quickest and easiest way was to completely uninstall AMD drivers and also delete the device directly so it would use the basic Windows drivers.
![imagen](https://github.com/user-attachments/assets/301afc98-889c-4458-b4d8-25b46f7b563c)

Once here, I performed a clean installation of 24.5.1 (very important, always check the reset to factory settings option).
![imagen](https://github.com/user-attachments/assets/235a715f-d4f0-4512-92db-41647a5d25b8)

And everything came back to life, LHM now correctly provides information about the GPU and FANCONTROL runs smoothly. Moreover, I believe even ADXL might work under version 24.5.1 because the message that ADXL cannot find the sensors no longer appears.

![Captura de pantalla 2024-07-18 214720](https://github.com/user-attachments/assets/77f3e3c3-9d1c-4d7d-8fe0-c77149e344c7)
And everything came back to life; LHM now correctly provides information about the GPU, and FANCONTROL runs smoothly. Although I use LHM to detect the GPU sensors, I would venture to say that the ADXL API might work under version 24.5.1. I mention this because the error message saying that ADXL cannot find the sensors does not appear when I activate its option.

@Rem0o, sorry for the long text and thank you very much again for everything , I hope this can help other colleagues.

**Comment by Kinetix288** [comment]
@lunaroc @Rem0o Hello just wanted to say I had the issue today as well on V247. I downgraded drivers because of a game and fan control couldnt recognize gpu thermal sensors and neither in LHMB. 

Uninstalling drivers with DDU and doing fresh amd adrenalin install with factory reset option resolved the issue completely. 

Adrenalin 25.9.2 and 7900 XT. Thank you for help!
